apps/src/aichat/api/validators.ts

```diff
@@ -0,0 +1,44 @@
+import {ResponseValidator} from '@cdo/apps/util/HttpClient';
+
+import {isChatMessage, ServerChatEvent} from '../types';
+
+/**
+ * Validates ChatEvents fetched by the user chat history API.
+ * Currently, this just checks for an ID, and ensures that chat messages are
+ * in the correct format.
+ * We may add more validation in the future to minimize the risk of breaking changes.
+ */
+const chatHistoryValidator: ResponseValidator<ServerChatEvent[]> = bodyJson => {
+  if (!Array.isArray(bodyJson)) {
+    throw new Error('Expected an array of chat events');
+  }
+
+  const events = bodyJson as ServerChatEvent[];
+  for (const event of events) {
+    if (event.id === undefined) {
+      throw fieldError('id');
+    }
+
+    if (isChatMessage(event)) {
+      for (const field of ['chatMessageText', 'role', 'status'] as const) {
+        if (event[field] === undefined) {
+          throw fieldError(field);
+        }
+      }
+
+      // Clear out assets if they were stored in the wrong format.
```

Comment:
Maybe an update to note why this is here (ie, format change pre-launch).

Reply:
Oh haha I was reviewing the commit with the most of the changes in it, looks like you updated this comment later  👍